gordon/schema/definitions/record-action.json

```diff
@@ -0,0 +1,36 @@
+{
+  "type": "object",
+  "$schema": "http://json-schema.org/draft-03/schema",
+
+  "title": "An action to perform on a DNS record.",
+  "description" : "Schema for creating or deleting selected DNS records.",
+  "properties": {
+    "name": {
+      "type": "string",
+      "required": true,
+      "examples": [
+        "fully-qualified-instance-name-1234.subdomain.example.com",
+        "www.example.com"
+      ]
+    },
+    "target": {
+      "type": "string",
+      "required": true,
+      "examples": [
+        "192.168.0.1",
+        "some-other-domain.example.com",
+        "some text value"
+      ]
+    },
+    "type": {
+      "type": "string",
+      "enum": ["A", "CNAME", "PTR", "TXT"],
+      "default": "A"
+    },
+    "action": {
+      "type": "string",
+      "enum": ["create", "delete"],
+      "default": "create"
+    }
+  }
+}
```

Comment:
@spotify/alf Did anyone have any feedback about the schema itself? I'm not sure if it's sufficient for what is needed (i.e. only certain types supported, one record = one message, on additional ttl or weights).  I wanted something sort of generic and not with more details than we need, but I'm not sure if I went to far.  I also don't know if we need a specific json spec if the input is just going to be parsing yaml to send off a json message to Gordon only to be reformatted back to a yaml record.

Reply:
I assume that this schema is for something like an event where we know the information for, i.e. we don't have to talk to GCE in order to get host information. If I have that wrong, lmk.

Assuming above, I think an "update" would be a good action, since a record could just be changed. I think it could be hard for our manual record event publishing based off of git diffs to submit two messages, one to delete and one to create, especially since there is no guaranteed ordering w/ google pubsub.

Reply:
What would you think of adding sub-schemas depending on record type that are based on google's resource record format: https://cloud.google.com/dns/records/json-record ?